parsers/FR_O.py

```diff
@@ -0,0 +1,277 @@
+# https://www.guadeloupe-energie.gp/wp-content/uploads/2011/07/2010-10-01_Biomasse_Etat-des-lieux.pdf p.35)
+MAP_GENERATION_BAGASSE_COAL_GUADELOUPE = {
+    1: {'coal': 1, 'biomass': 0},
+    2: {'coal': 1, 'biomass': 0},
+    3: {'coal': 0.6, 'biomass': 0.4},
+    4: {'coal': 0.3, 'biomass': 0.7},
+    5: {'coal': 0.3, 'biomass': 0.7},
+    6: {'coal': 0.6, 'biomass': 0.4},
+    7: {'coal': 0.8, 'biomass': 0.2},
+    8: {'coal': 1, 'biomass': 0},
+    9: {'coal': 1, 'biomass': 0},
+    10: {'coal': 1, 'biomass': 0},
+    11: {'coal': 1, 'biomass': 0},
+    12: {'coal': 1, 'biomass': 0}
+}
+
+# -------------- Thermal Repartitions -------------- #
+# These approximations are made from this document :
+# https://opendata-corse-outremer.edf.fr/api/datasets/1.0/courbe-de-charge-de-la-production-delectricite-par-filiere/attachments/bilan_electrique_edf_sei_2017_pdf/
+
+MAP_GENERATION_DIESEL_GAS_REUNION = {
+    'gas': 0.1202,
+    'oil': 0.8798
+}
+MAP_GENERATION_DIESEL_GAS_MARTINIQUE = {
+    'gas': 0.1247,
+    'oil': 0.8753
+}
+MAP_GENERATION_DIESEL_GAS_GUADELOUPE = {
+    'gas': 0.0126,
+    'oil': 0.9874
+}
+MAP_GENERATION_DIESEL_GAS_GUYANNE = {
+    'gas': 0.2261,
+    'oil': 0.7739
+}
+MAP_GENERATION_DIESEL_GAS_CORSE = {
+    'gas': 0.0196,
+    'oil': 0.9804
+}
+
+# Thermal sources by region
+thermal_mapping = {
+    'FR-COR': [
+        {
+            'name': 'thermique_mwh',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_CORSE
+        }
+    ],
+    'GF': [
+        {
+            'name': 'thermique_mwh',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_GUYANNE
+        }
+    ],
+    'GP': [
+        {
+            'name': 'bagasse_charbon_mwh',
+            'monthly_variation': True,
+            'type': MAP_GENERATION_BAGASSE_COAL_GUADELOUPE
+        },
+        {
+            'name': 'thermique_mwh',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_GUADELOUPE
+        }
+    ],
+    'MQ': [
+        {
+            'name': 'thermique_mwh',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_MARTINIQUE
+        }
+    ],
+    'RE': [
+        {
+            'name': 'bagasse_charbon_mwh',
+            'monthly_variation': True,
+            'type': MAP_GENERATION_BAGASSE_COAL_REUNION
+        },
+        {
+            'name': 'bagasse_charbon',
+            'monthly_variation': True,
+            'type': MAP_GENERATION_BAGASSE_COAL_REUNION
+        },
+        {
+            'name': 'thermique_mwh',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_REUNION
+        },
+        {
+            'name': 'thermique',
+            'monthly_variation': False,
+            'type': MAP_GENERATION_DIESEL_GAS_REUNION
+        }
+    ],
+}
+
+# Non-thermal sources
+sources = ['biomass', 'hydro', 'solar', 'wind', 'geothermal']
```

Comment:
Seems like this isn't really needed if you have `source mapping`

Reply:
Yes, moreover, the gas is missing here. This needs some cleaning :2nd_place_medal: